### PR TITLE
Add GitHub Actions support to publish images

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Adds GitHub Actions support to publish `mattermost-elasticsearch` images.

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Publish Docker image
+
+on:
+  release:
+    types: [published]
```
Comment (toninis) [inline .github/workflows/docker.yml]: We do not create releases . This should be on [push.tags](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-branches-and-tags)

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        id: meta
+        uses: docker/metadata-action@818d4b7b91585d195f67373fd9cb0332e31a7175 # v4.6.0
+        with:
+          images: mattermostdevelopment/mattermost-elasticsearch
+      
+      - name: cd/Build and push Docker image
+        uses: docker/build-push-action@2eb1c1961a95fc15694676618e422e8ba1d63825 # v4.1.1
+        with:
+          context: .
+          file: ./Dockerfile
```
Comment (toninis) [inline .github/workflows/docker.yml:31]: This is needed actually for multi-platform . We can check that 
```suggestion
          file: ./Dockerfile
          platforms: linux/amd64,linux/arm64/v8
```

**Review by toninis — CHANGES_REQUESTED** [review]

**Comment by isacikgoz** [comment]
Gentle ping @toninis

**Review by toninis — APPROVED** [review]

**Comment by isacikgoz** [comment]
@toninis so we will need to add `secrets.DOCKERHUB_DEV_USERNAME }}` and `secrets.DOCKERHUB_DEV_TOKEN }}` to this repository right? Or they are accessible organization wide?

**Comment by toninis** [comment]
> @toninis so we will need to add `secrets.DOCKERHUB_DEV_USERNAME }}` and `secrets.DOCKERHUB_DEV_TOKEN }}` to this repository right? Or they are accessible organization wide?

Already done . You are good to go 🚀

**Pull request merged**